chore: updating block loader

davidnuescheler opened the pull request:
only vaguely related, i think we should have a default fstab in here that points to example content (read only) and i think that we could allow copying which would make it easy for someone to get started.

rofe reviewed (changes requested):
LGTM, apart from the linting errors

davidnuescheler commented:
@rofe & @kptdobe i updated a few more things and changed the sequence, and i am not sure why the test fails. it may very well be that i am doing something wrong.

kptdobe commented on scripts/scripts.js:
```diff
@@ -122,6 +169,7 @@ function decorateBlocks($main) {
  * @param {string} blockName name of the block
  * @param {any} content two dimensional array or string or object of content
  */
+// eslint-disable-next-line no-unused-vars
```
Really needed ?

rofe replied:
This function is no longer used by default, but could theoretically be used to create synthetic/auto blocks. But for that purpose, it would need to be exported...

davidnuescheler replied:
i think this is a function that i think we should encourage to use, possibly supported with an example. i think one of the most typical examples would be a hero/marquee section that is turned into a hero block. i have come to believe that somehow the templates or page types have a close connection with autoblocking.
meaning that if you have eg. a `category` page or a `article` page or similar those would create different auto blocks from the default content. somehow feels like an intuitive concept but i am not sure how to marry the two...
i think for now we could possibly just use an `h1` with a preceding `picture` and turn that into a `marquee` block as an example.

kptdobe commented on fstab.yaml:
```diff
@@ -1,2 +1,2 @@
-#   /: <shared-folder-url>
\ No newline at end of file
+mountpoints:
+  /: https://drive.google.com/drive/u/0/folders/1MGzOt7ubUh3gu7zhZIPb7R7dyRzG371j
```
You probably want to revert this one...

rofe replied:
@kptdobe I think this is intentional and part of the new first run experience @davidnuescheler had in mind: immediately show some content and guide the developer on what to do next.

kptdobe replied:
ah yes, this even the PR description :)

kptdobe commented on scripts/scripts.js:
```diff
@@ -152,29 +200,43 @@ function buildBlock(blockName, content) {
-  $main
-    .querySelectorAll('div.section-wrapper > div > .block')
-    .forEach(async ($block) => loadBlock($block));
+export function loadBlocks($main) {
```
Missing the JS doc for  `return`.

davidnuescheler replied:
done

rofe reviewed (approved):
Add an `eslint-ignore-next-line` statement for the 2 `console.log` statements (or comment them out?)

rofe commented:
>  i am not sure why the test fails. it may very well be that i am doing something wrong.
Never mind that, i think we should switch to web test runner anyway.
